src/oop_ext/foundation/types_.py

```diff
@@ -87,47 +90,46 @@ def __getattr__(self, mname):
 
         return self
 
-    def __setattr__(self, _name, _value):
+    def __setattr__(self, _name: str, _value: object) -> Any:
         "Ignore attribute setting."
         return self
 
-    def __delattr__(self, _name):
+    def __delattr__(self, _name: str) -> None:
         "Ignore deleting attributes."
-        return self
 
-    def __enter__(self):
+    def __enter__(self) -> Any:
         return self
 
-    def __exit__(self, *args, **kwargs):
-        return self
+    def __exit__(self, *args: object, **kwargs: object) -> None:
+        pass
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         "Return a string representation."
         return "<Null>"
 
-    def __str__(self):
+    def __str__(self) -> str:
         "Convert to a string and return it."
         return "Null"
 
-    def __bool__(self):
+    def __bool__(self) -> "Literal[False]":
```

Comment:
I didn't get the usage of `Literal` here, this annotation would not be only ` -> bool`?

Reply:
No, it literally always returns `False`. This allows mypy to be smart and infer what this returns.

For example:

```python
n = Null()
if n:
    # mypy can tell this block is unreachable.
```